vizalerts.py

```diff
@@ -615,18 +616,38 @@ def process_csv(csvpath, view, sitename, viewname, subscriberemail, subscribersy
             data = get_unique_vizdata(data, has_consolidate_email, has_email_from, has_email_cc, has_email_bcc, has_email_header, has_email_footer)
             rowcount_unique = len(data)
 
-            # determine whether we need to render the viz image--if so, render it to file
-            vizimagerefs = find_vizimages_in_body(data, has_email_header, has_email_footer)
-            if vizimagerefs:
+            
+            # could be multiple viz images for a single row in the CSV
+            # return a list of all found VIZ_IMAGE() strings including any custom views w/ or w/out URL parameters
+            # leaving them as VIZ_IMAGE() strings because it makes the search & replace easier for the body & attachments
+            vizimagerefs = []
+            vizimagerefs = find_vizimages_in_body(data, viewurlsuffix, has_email_header, has_email_footer)
+
+            # storing view in original value because we'll be rewriting it in order to call
+            # tabhttp.export_view
+            origviewsuffix = view['view_url_suffix']
+
+            
+            # creating a dictionary of imagepaths with vizimageref as the key
+            imagepaths = dict()
+            for vizimageref in vizimagerefs:
+                if vizimageref == IMAGE_PLACEHOLDER:
+                    view['view_url_suffix'] = origviewsuffix
+                else:
+                    view['view_url_suffix'] = re.match(u'.*?\(\'(.*?)\'\)', vizimageref).group(1)
```

Comment:
If I'm reading the regex right, this is looking for the first open paren followed by a quote character: (', then the reverse: ')
Per our email discussion, I think we're better off leaving the quotes out since it'll be complex / confusing / annoying for users to escape them in their Tableau calculations. We can use the pipe | character to break apart any additional arguments in the VIZ_IMAGE() function.

I realize now too, we should inform users that they can't use parentheses in their view paths within the function, as while they're valid characters in a URL, they'd screw up our ability to parse the boundaries of the function. That's pretty edge-casey though.